test/framework/src/main/java/org/elasticsearch/test/AbstractQueryTestCase.java

```diff
@@ -308,7 +308,7 @@ public void testUnknownField() throws IOException {
         } while (testQuery.toString().contains(marker));
         testQuery.queryName(marker); // to find root query to add additional bogus field there
         String queryAsString = testQuery.toString().replace("\"" + marker + "\"", "\"" + marker + "\", \"bogusField\" : \"someValue\"");
-        ParsingException e = expectThrows(ParsingException.class, () -> parseQuery(queryAsString));
+        RuntimeException e = expectThrows(RuntimeException.class, () -> parseQuery(queryAsString));
```

Comment:
That's a good point. Now I think ParsingExceptions would be better here. I think we should wrap the IllegalArgumentException that the object parsers throw currently in a ParsingException in fromXContent() then.